File: api/src/main/java/net/blueshell/api/mapper/SponsorMapper.java
package net.blueshell.api.mapper;

import net.blueshell.api.base.BaseMapper;
import net.blueshell.api.dto.SponsorDTO;
import net.blueshell.api.model.Sponsor;
import org.mapstruct.BeanMapping;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;
import org.mapstruct.NullValuePropertyMappingStrategy;

@Mapper(componentModel = "spring")
public abstract class SponsorMapper extends BaseMapper<Sponsor, SponsorDTO> {

    @BeanMapping(ignoreByDefault = true)
    public abstract Sponsor fromDTO(SponsorDTO dto, @MappingTarget Sponsor sponsor);

    @BeanMapping(ignoreByDefault = true)
    public abstract SponsorDTO toDTO(Sponsor sponsor);
}

File: api/src/main/kotlin/net/blueshell/api/mapper/SponsorMapper.kt
package net.blueshell.api.mapper

import net.blueshell.api.base.BaseMapper
import net.blueshell.api.dto.SponsorDTO
import net.blueshell.api.model.Sponsor
import org.mapstruct.BeanMapping
import org.mapstruct.Mapper
import org.mapstruct.MappingTarget

@Mapper(componentModel = "spring")
abstract class SponsorMapper : BaseMapper<Sponsor, SponsorDTO>() {
    @BeanMapping(ignoreByDefault = true)
    abstract fun fromDTO(dto: SponsorDTO, @MappingTarget sponsor: Sponsor): Sponsor

    @BeanMapping(ignoreByDefault = true)
    abstract override fun toDTO(sponsor: Sponsor): SponsorDTO
}
